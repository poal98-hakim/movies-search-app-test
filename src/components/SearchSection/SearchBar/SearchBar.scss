@use '../../../scss/base/colors';

.search-bar-wrapper{
    width: 20em;
    padding: 10px 0;
    border-bottom: 2px solid;
    color: colors.$PRIMARY_COLOR;

    input {
        border: 0;
        outline: none;
        color: colors.$PRIMARY_COLOR;
        font-size: 1.2em;
        margin-left: 10px;
        font-weight: 900;
        background-color: transparent;

        &::placeholder {
            opacity: .8;
            color: colors.$PRIMARY_COLOR;
            font-weight: 300;
        }
    }

}