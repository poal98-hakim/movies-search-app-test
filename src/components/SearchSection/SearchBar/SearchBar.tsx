import React from "react";
import "./SearchBar.scss";

interface IProps{
  icon: {src: string, alt: string};
  placeholder: string;
  value: string;
  onChange: (keyword: string) => void;
  type: string;
}
const SearchBar : React.VFC<IProps> = ({ icon, placeholder, onChange, type, value }) => {
  return (
    <div className="search-bar-wrapper flex f-align-center f-justify-center" data-testid="search-bar">
      <img src={icon.src} alt={icon.alt} width="25" />
      <input type={type} value={value} onChange={e => onChange(e.target.value)} placeholder={placeholder} />
    </div>
  );
}

export default SearchBar;
