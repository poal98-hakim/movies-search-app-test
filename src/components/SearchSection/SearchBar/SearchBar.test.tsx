import SearchBar from "./SearchBar";
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

const PROPS = {
  icon: {alt: "search icon", src: "http://localhost/search-icon-yellow.png"},
  type: "text",
  placeholder: "Search",
  value: "test",
  onChange: jest.fn()
}

describe('SearchBar', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<SearchBar {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<SearchBar {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads search bar', () => {
    expect(screen.getByTestId("search-bar")).toBeInTheDocument();
  })

  it('shows image', () => {
    expect(screen.getByAltText(props.icon.alt)).toBeInTheDocument();
    expect(screen.getByRole("img")).toHaveAttribute("src", props.icon.src);
  })

  it('loads correct input', () => {
    const searchBar = screen.getByRole("textbox");
    expect(searchBar).toBeInTheDocument();
    expect(searchBar).toHaveAttribute("type", props.type);
    expect(searchBar).toHaveAttribute("placeholder", props.placeholder);
    expect(searchBar).toHaveValue(props.value);
  })

  it('fires handler on changing input', () => {
    userEvent.type(screen.getByRole("textbox"), "10");
    expect(props.onChange).toHaveBeenCalled();
  })


})
