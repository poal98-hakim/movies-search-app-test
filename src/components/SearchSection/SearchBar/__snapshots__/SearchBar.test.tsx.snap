// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<body>
  <div>
    <div
      class="search-bar-wrapper flex f-align-center f-justify-center"
      data-testid="search-bar"
    >
      <img
        alt="search icon"
        src="http://localhost/search-icon-yellow.png"
        width="25"
      />
      <input
        placeholder="Search"
        type="text"
        value="test"
      />
    </div>
  </div>
  <div>
    <div
      class="search-bar-wrapper flex f-align-center f-justify-center"
      data-testid="search-bar"
    >
      <img
        alt="search icon"
        src="http://localhost/search-icon-yellow.png"
        width="25"
      />
      <input
        placeholder="Search"
        type="text"
        value="test"
      />
    </div>
  </div>
</body>
`;
